### ISSUE TITLE
How to use regularization for a feed forward neural network in R using mxnet? 

### ISSUE THREAD

**Issue by antonysamuelb** [apache/incubator-mxnet#7127]
## Environment info
Operating System: Windows

Package used: R (Installed using the binaries.)

Which parameter should I use to do regularization (L2) for a feed forward neural network?
Where should I implement this. 

I could find references to the weight decay parameter (wd). But I am not sure where to include it. 

An example would be much appreciated.

**Comment by thirdwing** [comment]
I think weight decay is what you are looking for.

A small example is given below. You can just add `wd` in `mx.model.FeedForward.create`.

```r
require(mlbench)
require(mxnet)

data(BostonHousing, package = "mlbench")

train.ind <- seq(1, 506, 3)
train.x <- data.matrix(BostonHousing[train.ind,-14])
train.y <- BostonHousing[train.ind, 14]
test.x <- data.matrix(BostonHousing[-train.ind,-14])
test.y <- BostonHousing[-train.ind, 14]

data <- mx.symbol.Variable("data")
fc1 <- mx.symbol.FullyConnected(data, num_hidden = 1)
lro <- mx.symbol.LinearRegressionOutput(fc1)

mx.set.seed(0)
model <- mx.model.FeedForward.create(lro,
                                     X = train.x,
                                     y = train.y,
                                     ctx = mx.cpu(),
                                     num.round = 50,
                                     array.batch.size = 20,
                                     optimizer = "sgd",
                                     learning.rate = 2e-6,
                                     momentum = 0.9,
                                     wd = 0.1,
                                     eval.metric = mx.metric.rmse)
```

Sorry for the lack of document.
